### PR TITLE
Add ESM support

### PR DESCRIPTION
**Pull request by giuseppeg**
Add ESM support for the legacy compiler (no Vite).

Replaces #29

### CODE REVIEW

**Comment by giuseppeg** [comment]
@itsMapleLeaf it would be great if you could add your formatter configuration to the repo such that formatting is consistent otherwise the script will probably use my personal config

File: README.md [hunk 1]
```diff
@@ -94,31 +94,33 @@ Initializes remix-electron. Returns a promise with a url to load in the browser
 
 Options:
 
-- `serverBuild`: The path to your server build (e.g. `path.join(__dirname, 'build')`), or the server build itself (e.g. required from `@remix-run/dev/server-build`). Updates on refresh are only supported when passing a path string.
+- `serverBuild`: The path to your server build (e.g. `path.join(process.cwd(), 'build')`), or the server build itself (e.g. required from `@remix-run/dev/server-build`). Updates on refresh are only supported when passing a path string.
```
Comment (itsMapleLeaf) [inline README.md:97]: Why the change here? I figure `__dirname` or even `import.meta.url` would be a more reliable reference path

**Comment by itsMapleLeaf** [comment]
I'll take a closer look at this later, thanks for the PR!

File: README.md (hunk 1 above)
Comment (giuseppeg) [reply README.md:97]: I think that `__dirname` is not available in ESM, `import.meta.url` should do but we'd need to document both. Happy to make this change if you prefer that but `process.cwd()` should fine I think

**Comment by giuseppeg** [comment]
@itsMapleLeaf thanks! If you add the formatter config on `main` I will rebase and format again, so it'll be easier to go through the changes.

File: workspaces/remix-electron/src/index.mts [hunk 2]
```diff
@@ -38,18 +39,29 @@ export async function initRemix({
 	mode,
 	publicFolder: publicFolderOption = "public",
 	getLoadContext,
+	esm = typeof require === "undefined",
 }: InitRemixOptions): Promise<string> {
-	const appRoot = app.getAppPath()
```
Comment (giuseppeg) [inline workspaces/remix-electron/src/index.mts:42]: this is the electron app root not the remix project's and therefore `publicFolder` will be wrong when resolved below, so I replaced it with `process.cwd()`

File: README.md (hunk 1 above)
Comment (itsMapleLeaf) [reply README.md:97]: Yeah it doesn't make a huge difference, I've just been bitten by `process.cwd()` in the past

**Comment by itsMapleLeaf** [comment]
> @itsMapleLeaf thanks! If you add the formatter config on `main` I will rebase and format again, so it'll be easier to go through the changes.

It sounds like you're asking for the Prettier config? This project uses Biome, not Prettier, so you would need to install the Biome extension https://biomejs.dev/reference/vscode/

**Comment by giuseppeg** [comment]
> It sounds like you're asking for the Prettier config? This project uses Biome, not Prettier, so you would need to install the Biome extension https://biomejs.dev/reference/vscode/

Thanks for clarifying! The problem is that I don't use VSCode I guess. Not sure how to solve it, except for running `pnpm format`, which I did... perhaps the Biome has something like .prettierrc such that `pnpm format` will use that config?

**Comment by itsMapleLeaf** [comment]
Biome's config is all in `biome.json`. The format script seems to work fine on my end, so uh... 😅 I'm not really sure! I'll try rebasing this again, but manually.

EDIT: So the formatter action works! But I think the issue here is that Biome doesn't support markdown yet 😢 

I think I'm gonna switch back to Prettier until it does. Big miss on my part 😅 

There are some new conflicts, they shouldn't be too difficult to address. I think I'll leave those up to you in case there's anything else you want to do here.

**Comment by giuseppeg** [comment]
Thank you @itsMapleLeaf and sorry for being a bit slow to reply. I rebased and checks seem to be all green now.

**Review by itsMapleLeaf — APPROVED** [review]
